File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-3.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int currMax = -1001;
    public int MaxPathSum(TreeNode root) {
        TraverseNodes(root);
        return currMax;
    }

    private int TraverseNodes(TreeNode curr)
    {
        if (curr == null)
            return 0;

        int ls = TraverseNodes(curr.left) + curr.val;
        int rs = TraverseNodes(curr.right) + curr.val;

        int combined = ls + rs - curr.val;

        int max = Math.Max(ls, rs);

        max = Math.Max(max, curr.val);

        int ckMax = Math.Max(max, combined);
        currMax = Math.Max(ckMax, currMax);

        return max;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    levels = []
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        self.levels = []
        self.addToLevels(root, 0)

        
        return self.levels

    def addToLevels(self, curr: Optional[TreeNode], level: int):

        if curr is None:
            return
        
        if len(self.levels) <= level:
            self.levels.append([])

        self.levels[level].append(curr.val)

        level += 1

        self.addToLevels(curr.left, level)
        self.addToLevels(curr.right, level)